palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/request/OriginalRequest.java

```diff
@@ -0,0 +1,144 @@
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.util.Assert;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.Generated;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+
+
+/**
+ * Represents the original data that has been sent from the client to Palisade Service for a request to access data.
+ * This data will be forwarded to a set of services with each contributing to the processing of this request.
+ * This version represents the original request.
+ * Next in the sequence is the input for user-service where this data will be used as a request for a User.
+ * Note there are two classes that effectively represent the same data but represent a different stage of the process.
+ * uk.gov.gchq.palisade.service.palisade.request.OriginalRequest is the client request that has come into the Palisade Service.
+ * uk.gov.gchq.palisade.service.user.request.UserRequest is the input for the User Service.
+ */
+
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public final class OriginalRequest {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private final String userId;  //Unique identifier for the user
+    private final String resourceId;  //Resource that that is being asked to access
+    private final JsonNode context; //Relevant context information about the request.
+
+
+    @JsonCreator
+    private OriginalRequest(
+            final @JsonProperty("userId") String userId,
+            final @JsonProperty("resourceId") String resourceId,
+            final @JsonProperty("context") JsonNode context) {
+
+        Assert.notNull(userId, "User cannot be null");
+        Assert.notNull(resourceId, "Resource cannot be null");
+        Assert.notNull(context, "Context cannot be null");
+
+        this.userId = userId;
+        this.resourceId = resourceId;
+        this.context = context;
+    }
+
+    @Generated
+    public String getUserId() {
+        return userId;
+    }
+
+    @Generated
+    public String getResourceId() {
+        return resourceId;
+    }
+
+    @Generated
+    public Context getContext() throws JsonProcessingException {
+        return MAPPER.treeToValue(context, Context.class);
+    }
+
+    @Override
+    @Generated
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof OriginalRequest)) {
+            return false;
+        }
+        OriginalRequest that = (OriginalRequest) o;
+        return userId.equals(that.userId) &&
+                resourceId.equals(that.resourceId) &&
+                context.equals(that.context);
+    }
+
+    @Override
+    @Generated
+    public int hashCode() {
+        return Objects.hash(userId, resourceId, context);
+    }
```

Comment:
Do we need this?

Reply:
It is part of standard convention so I will first assume it needs to be there and then remove it when I can convince anyone that it should not be there.